infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/WaitSnapshotApplyState.java

```diff
@@ -110,6 +118,12 @@ public LogReplicationState processEvent(LogReplicationEvent event) throws Illega
                     logEntrySyncState.setTransitionEventId(event.getEventId());
                     fsm.setBaseSnapshot(event.getMetadata().getLastTransferredBaseSnapshot());
                     fsm.setAckedTimestamp(event.getMetadata().getLastLogEntrySyncedTimestamp());
+                    if (tableManagerPlugin.isUpgraded()) {
+                        // If LR is in upgrading path, it means this cycle of snapshot sync was triggered
+                        // forcibly because LR detected a version mismatch. Flipping the flag back to false
+                        // here to indicate that the upgrade path is completed.
+                        tableManagerPlugin.updateUpgradeFlag(false);
```

Comment:
this update must be done atomically alongwith the metadata updates in L119 and L120

Reply:
please ignore this comment.  Just realized that the above 2 updates are in-memory and not updating any tables.